content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md

```diff
@@ -178,6 +192,313 @@ Error from server (NotFound): Unable to list {"stable.example.com" "v1" "crontab
+
+
+Example 1:
+```yaml
+allOf:
+- properties:
+    foo:
+      ...
+```
+conflicts with rule 2. The following would be correct:
+```yaml
+properties:
+  foo:
+    ...
+allOf:
+- properties:
+    foo:
+      ...
+```
+
+Example 2:
+```yaml
+allOf:
+- items:
+    properties:
+      foo:
+        ...
+```
+conflicts with rule 2. The following would be correct:
+```yaml
+items:
+  properties:
+    foo:
+      ...
+allOf:
+- items:
+    properties:
+      foo:
+        ...
+``` 
+
+Example 3:
+```yaml
+properties:
+  foo:
+    pattern: "abc"
+  metadata:
+    type: object
+    properties:
+      name:
+        type: string
+        pattern: "^a"
+      finalizers:
+        type: array
+        items:
+          type: string
+          pattern: "my-finalizer"
+anyOf:
+- properties:
+    bar:
+      type: integer
+      minimum: 42
+  required: ["bar"]
+  description: "foo bar object"
+```
+is not a structural schema because of the following violations:
+
+* the type at the root is missing (rule 1).
+* the type of `foo` is missing (rule 1).
+* `bar` inside of `anyOf` is not specified outside (rule 2). 
+* `bar`'s `type` is within `anyOf` (rule 3).
+* the description is set within `anyOf` (rule 3).
+* `metadata.finalizer` might not be restricted (rule 4). 
+
+In contrast, the following, corresponding schema is structural:
+```yaml
+type: object
+description: "foo bar object"
+properties:
+  foo:
+    type: string
+    pattern: "abc"
+  bar:
+    type: integer
+  metadata:
+    type: object
+    properties:
+      name:
+        type: string
+        pattern: "^a"
+anyOf:
+- properties:
+    bar:
+      minimum: 42
+  required: ["bar"]
+```
+
+Violations of the structural schema rules are reported in the `NonStructural` condition in the CustomResourceDefinition.
+
+Not being structural will disable the following features:
+
+* [Validation Schema Publishing](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#publish-validation-schema-in-openapi-v2)
+* [Webhook Conversion](/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#webhook-conversion)
+* [Validation Schema Defaulting](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#defaulting)
+* [Pruning](#preserving-unknown-fields)
+
+and possibly more features in the future.
+
+### Pruning versus preserving unknown fields
+
+{{< feature-state state="beta" for_kubernetes_version="1.15" >}}
+
+CustomResourceDefinitions traditionally store any (possibly validated) JSON as is in etcd. This means that unspecified fields (if there is a [OpenAPI v3.0 validation schema](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#validation) at all) will be persisted. This is in contrast to native Kubernetes resources like e.g. a pod. If the user specifies a field `spec.randomField`, this will be accepted by the API server HTTPS endpoint.
```

Comment:
Isn't `If the user specifies a field `spec.randomField`, this will be accepted by the API server HTTPS endpoint.` missing the crucial detail "... but will be dropped before being persisted in etcd."?

Reply:
fixed